File: tests/fixtures/sample-python-project/services/user_service/handlers/user_handler.py
from flask import request, jsonify
from ..services.user_service import UserService
from shared.utils.validators import validate_email

user_service = UserService()


def get_users():
    users = user_service.get_all()
    return jsonify(users)


def get_user_by_id(user_id: str):
    user = user_service.get_by_id(user_id)
    if not user:
        return jsonify({"error": "Not found"}), 404
    return jsonify(user)


def create_user():
    data = request.get_json()
    name = data["name"]
    email = data["email"]
    if not validate_email(email):
        return jsonify({"error": "Invalid email"}), 400
    user = user_service.create({"name": name, "email": email})
    return jsonify(user), 201


def delete_user(user_id: str):
    user_service.delete(user_id)
    return "", 204

File: tests/fixtures/sample-csharp-project-negative/services/UserService/Handlers/UserHandler.cs
using Shared.Utils;
using UserServiceApp.Services;

namespace UserServiceApp.Handlers;

public static class UserHandler
{
    private static readonly UserServiceApp.Services.UserService _userService = new();

    public static IResult GetUsers()
    {
        var users = _userService.GetAll();
        return Results.Ok(users);
    }

    public static IResult GetUserById(string userId)
    {
        var user = _userService.GetById(userId);
        if (user == null)
        {
            return Results.NotFound(new { error = "Not found" });
        }
        return Results.Ok(user);
    }

    public static IResult CreateUser(Dictionary<string, object> data)
    {
        var name = data["name"].ToString()!;
        var email = data["email"].ToString()!;
        if (!Validators.ValidateEmail(email))
        {
            return Results.BadRequest(new { error = "Invalid email" });
        }
        var user = _userService.Create(new Dictionary<string, object>
        {
            ["name"] = name,
            ["email"] = email
        });
        return Results.Created($"/users/{user["id"]}", user);
    }

    public static IResult DeleteUser(string userId)
    {
        _userService.Delete(userId);
        return Results.NoContent();
    }
}
